projects/ngx-clipboard/src/lib/ngx-clipboard.directive.ts

```diff
@@ -16,12 +18,18 @@ export class ClipboardDirective implements OnInit, OnDestroy {
     @Input()
     public cbContent: string;
 
+    @Input()
+    public cbSuccessMsg: string;
+
     @Output()
-    public cbOnSuccess: EventEmitter<any> = new EventEmitter<any>();
+    public cbOnSuccess: EventEmitter<IClipboardResponse> = new EventEmitter<IClipboardResponse>();
 
     @Output()
     public cbOnError: EventEmitter<any> = new EventEmitter<any>();
-    constructor(private clipboardSrv: ClipboardService) {}
+    constructor(
+        private clipboardSrv: ClipboardService,
+        @Inject(ClipboardResponseService) private responseService: ClipboardResponseService
```

Comment:
not sure about the syntax here. Why don't use just use `private responseService: ClipboardResponseService`?

Reply:
yeah, forgot to remove @Inject(), an abstract class can directly act as injection token. thx

Reply:
not exposing Subject because of the reason you stated, which makes sense.

Reply:
That is not exactly what I meant.  Sorry for confusing you.  I meant we should expose it as observable. `subject.asObservable()`. Even thought changing type is enough because of type checking from typescript, I do prefer more explicit way.
```
private subject = new Subject<{}>(); 
public observable$: Observable<{}> = this.subject;
```

Reply:
aah! Right. This is much better, simple and straightforward.

Reply:
how do I get around the fact that the subject might still be accessible from other public methods?

Reply:
The problem I am facing is that copyFromInputElement method currently has no param defined for Event object and success message. So either I update their params thus allowing them to call the private Subject, or I create a method in the service itself which will push next values to the private Subject. Later will lead to user being able to uncontrollably call the Subject indirectly, though it will be guarded against error and complete.

Reply:
I think allowing users to push in a separate public method is fine. We don't expose subject because they might accidentally close it or something else, but pushing next in a dedicated should be ok.

Reply:
updated PR